src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao;
+
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.models.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+import java.util.Optional;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    private static final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw  new DataProcessingException("Can't add movie " + movie + " to DB.", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        Movie movie = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            movie = session.get(Movie.class, id);
+        } catch (HibernateException e) {
+            throw new DataProcessingException("Can't get movie wiht id = " + id + " from DB", e);
+        } finally {
+            session.close();
+        }
+        return Optional.ofNullable(movie);
```

Comment:
- use try with recourses
 - return Optioanal.ofNullable(session.get(Movie.class, id));
 - that's not a mistake, but u won't get an exception if there is no movie with such id